hack/scripts/validate_deployment.sh

```diff
@@ -0,0 +1,276 @@
+      if [[ $? -ne 0 ]]; then
+        echo -e "$ERROR $TEST_NAME failed.  Unable to create cluster."
+        exit 1
+      fi
+
+      # get the cluster ID from the OCM output
+      local CLUSTER_ID=$(echo $OCM_CREATE | grep '^ID:' | awk '{print $2}')
+
+      echo -e "$PENDING - CCS $CLUSTER_NAME created. Cluster ID $CLUSTER_ID"
+
+      ocm_cluster_test $CLUSTER_ID 1
+      ;;
+    sts)
+      local CLUSTER_NAME="aao-sts-$DATETIME"
+      ;;
+    *)
+      echo -e "$ERROR $TEST_NAME is not a valid test case" >2
+      exit 1
+      ;;
+  esac
+
+  echo -e " $SUCCESS $TEST_NAME is complete."
+}
+
+install_wait_loop()
+{
+  local CLUSTER_ID=$1
+  local TIMEOUT=$2
+  local CHECK_INTERVAL=$3
+  # Loop 
+  while true; do
+    sleep $CHECK_INTERVAL
+    TIMEOUT=$(( TIMEOUT - CHECK_INTERVAL ))
+    local STATUS=$(ocm get /api/clusters_mgmt/v1/clusters/$CLUSTER_ID/status | jq -r .state)
+    if [[ $STATUS == "installing" ]]; then
+      echo -e "$SUCCESS $TEST_NAME has passed. Beginning Cleanup..."
+      break
+    fi
+
+    if [[ $TIMEOUT -le 0 ]]; then
+      echo -e "$RED$BULLET$RESET $TEST_NAME has timed out.  Beginning Cleanup..."
+      break
+    fi
+
+    if [[ $STATUS == "pending" ]]; then
+      echo -e "$PENDING $TEST_NAME is still $STATUS..."
+    fi
+  done
+}
+
+delete_wait_loop()
+{
+  local CLUSTER_ID=$1
+  local TIMEOUT=$2
+  local CHECK_INTERVAL=$3
+  # Begin Cleanup Loop
+  while true; do
+    sleep $CHECK_INTERVAL
+    TIMEOUT=$(( TIMEOUT - CHECK_INTERVAL ))
+    local STATUS=$(ocm get /api/clusters_mgmt/v1/clusters/$CLUSTER_ID/status 2>&1)
+
+    if [[ $? -ne 0 ]]; then
+      echo -e "$SUCCESS $TEST_NAME has successfully finished cleaning up."
+      break
+    fi
+
+    # Sometimes the above command doesn't work, so let's also check the status that comes back in stderr
+    local STATUS_KIND=$(jq -r .kind <<< $STATUS)
+    local STATUS_ID=$(jq -r .id <<< $STATUS)
+
+    if [[ $STATUS_KIND == "Error" ]] && [[ $STATUS_ID == "404" ]]; then
+      echo -e "$SUCCESS $TEST_NAME has successfully finished cleaning up."
+      break
+    fi
+
+    # If we hit our timeout, let's exit.
+    if [[ $TIMEOUT -le 0 ]]; then
+      echo -e "$ERROR $TEST_NAME cleanup has timed out.  This could mean that resources are abandoned in your account and require manual cleanup."
+      break
+    fi
+
+    # Check to see if we're still uninstalling and let the user know
+    local STATE=$(jq -r .state <<< $STATUS)
+
+    if [[ $STATE == "uninstalling" ]]; then
+      echo -e "$PENDING $TEST_NAME is still $STATE..."
+    fi
+  done  
+}
+
+ocm_cluster_test()
+{
+  echo -e "$PENDING $TEST_NAME Started."
+  local CLUSTER_ID=$1
+
+  # Set the timeout to be 900 seconds, or 15m
+  local CREATE_TIMEOUT=900
+  local DELETE_TIMEOUT=900
+  
+  # Check Interval is how long between checks we go, in seconds
+  local CHECK_INTERVAL=5
+  
+  install_wait_loop $CLUSTER_ID $CREATE_TIMEOUT $CHECK_INTERVAL
+
+  ocm delete cluster $CLUSTER_ID
```

Comment:
how paranoid do we want to be about ensuring tests clean up after themselves? You may want to consider using a trap (https://www.linuxjournal.com/content/bash-trap-command) to ensure clean up happens even in script failure scenarios. That may require more engineering than is necessary though.

Reply:
Yeah, I was thinking about that, but then like you mentioned the engineering into that started to yak-shave.

Like, I could add the trap, but then I'd need to know whether or not the clusters are initiating already or not and then clean it up.  And then catch it in the subprocesses, and ignore it in the delete wait loop, so I just opted out of doing it at all.

It might be worth adding a simple trap though that just lets you know that depending on the state of each thread you might have to manually clean something up.